core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/HDFSAction.java

```diff
@@ -0,0 +1,140 @@
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * Move file(s) within HDFS.
+ * A user must specify file/directory path and destination file/directory path
+ * Optionals include fileRegex
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("HDFSAction")
+@Description("Action to move files within HDFS")
+public class HDFSAction extends Action {
+  private static final Logger LOG = LoggerFactory.getLogger(HDFSAction.class);
+
+  private HDFSActionConfig config;
+
+  public HDFSAction(HDFSActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    Path source = new Path(config.sourcePath);
+    Path dest = new Path(config.destPath);
+    FileSystem fileSystem = source.getFileSystem(new Configuration());
+
+    if (fileSystem.getFileStatus(source).isFile()) { //moving single file
+
+      try {
+        fileSystem.rename(source, dest);
+      } catch (IOException e) {
+        if (!config.continueOnError) {
+          e.printStackTrace();
+        }
+        LOG.error("Failed to move file {} to {} for reason: {}", source.toString(), dest.toString(), e);
+      }
+      return;
+    }
+
+    //moving contents of directory
+    FileStatus[] listFiles;
+    if (config.fileRegex != null) {
+      PathFilter filter = new PathFilter() {
+        private final Pattern pattern = Pattern.compile(config.fileRegex);
+
+        @Override
+        public boolean accept(Path path) {
+          return pattern.matcher(path.getName()).matches();
+        }
+      };
+
+      listFiles = fileSystem.listStatus(source, filter);
+    } else {
+      listFiles = fileSystem.listStatus(source);
+    }
+
+    for (FileStatus file: listFiles) {
+      source = file.getPath();
+      //Moving directory, so destination filePath doesn't have filename
+      dest = new Path(config.destPath);
+
+      try {
+        fileSystem.rename(source, dest);
+      } catch (IOException e) {
+        if (!config.continueOnError) {
+          e.printStackTrace();
+        }
+        LOG.error("Failed to move file {} to {} for reason: {}", source.toString(), dest.toString(), e);
+      }
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+
+  }
+
+  /**
+   * Config class that contains all properties necessary to execute an HDFS move command.
+   */
+  public class HDFSActionConfig extends PluginConfig {
+    @Description("Full HDFS path of file/directory")
+    private String sourcePath;
+
+    @Description("Full HDFS path of desired destination path")
+    private String destPath;
+
+    @Nullable
+    private String fileRegex;
+
+    private boolean continueOnError;
+
+    HDFSActionConfig() {
+      continueOnError = false;
```

Comment:
its not optional, so the user is forced to set it, so there is no point in a default. If it made optional, then this is the correct way to set the default.